apps/server/src/modules/tool/service/tool-validation.service.spec.ts

```diff
@@ -118,6 +118,30 @@ describe('ToolValidation', () => {
 			});
 		});
 
+		describe('validateDefaultValue', () => {
+			describe('when default value matches regex', () => {
+				it('should not throw', async () => {
+					const externalToolDO: ExternalToolDO = externalToolDOFactory
+						.withCustomParameters(1, { default: 'test', regex: '[t]', regexComment: 'mockComment' })
+						.buildWithId();
+
+					const result: Promise<void> = service.validateCreate(externalToolDO);
+
+					await expect(result).resolves.not.toThrow();
+				});
+
+				it('should throw', async () => {
```

Comment:
this probabably is in the wrong place: "when default value matches regex, it should throw"
Hint, because I see this a lot: when you only read the test descriptions, and they tell you the code should behave differently under the same condition, something is off.